utils/training_data_utils.py

```diff
@@ -1,221 +1,148 @@
+    return cell_polygon.contains(point)
 
 
 def get_labeled_cells(
-    features_path: pathlib.Path,
-    training_set_dat_path: pathlib.Path,
-    training_data: pd.DataFrame,
+    training_data: pd.DataFrame, features_samples_path: pathlib.Path
 ) -> pd.DataFrame:
     """
-    get labeled cells from larger training dataset
+    get labeled cells as dataframe from all training data and features samples
 
     Parameters
     ----------
-    features_path : pathlib.Path
-        path to features directory
-    training_set_dat_path : pathlib.Path
-        path to trainingset file
     training_data : pd.DataFrame
-        larger training dataset
+        all single cell features from all frames with any labeled cells
+    features_samples_path : pathlib.Path
+        path to features samples file
 
     Returns
     -------
     pd.DataFrame
-        dataset with all labeled single cells, including mitocheck-assigned cell IDs and phenotypic class
+        dataframe with all labeled cells
     """
-    labeled_cells = []
-
-    for index, row in training_data.iterrows():
-        centroid = (row["Location_Center_X"], row["Location_Center_Y"])
-        plate = row["Metadata_Plate"]
-        well = row["Metadata_Well"]
-        frame = row["Metadata_Frame"]
-        try:
-            frame_features_path = get_frame_features_path(
-                features_path, plate, well, frame
-            )
-            frame_features = pd.read_csv(
-                frame_features_path, compression="gzip", header=None
+    with open(features_samples_path) as labels_file:
+        labeled_cells = []
+
+        # iterate through each sample and see if it has features in the total training data dataframe
+        for line in labels_file:
+            # get phenotpic label of cell from feature samples file line
+            phenotypic_class = line.strip().split("\t")[0]
+            # getframe info from feature samples file line
+            frame_details = line.strip().split("\t")[1]
+            plate, well_num, frame, center_x, center_y = get_frame_metadata(
+                frame_details
             )
-            frame_labels = get_frame_labels(training_set_dat_path, plate, well, frame)
-            obj_id, cell_is_labeled = is_labeled(centroid, frame_features, frame_labels)
-
-            if cell_is_labeled:
-                phenotypic_class = get_cell_class(
-                    training_set_dat_path, plate, well, frame, obj_id
+            # get all single cell features for this particular frame
+            frame_cells = training_data.loc[
+                (training_data["Metadata_Plate"] == plate)
+                & (training_data["Metadata_Well"] == str(well_num))
+                & (training_data["Metadata_Frame"] == str(frame))
+            ]
+
+            included = False
+            # see if the center coords correspond to any feature data from the frame cells
+            for index, row in frame_cells.iterrows():
```

Comment:
Since you don't use index from `iterrows()` here, I'd recommend replacing the variable name with `_` which is commonly used in Python for referencing "throwaway variables".

```suggestion
            for _, row in frame_cells.iterrows():
```

Reply:
Thank you! Done in https://github.com/WayScience/mitocheck_data/pull/23/commits/12a1b200d1a6a9d04954f1f019ef86046e91cbcf